buildspec.yaml

```diff
@@ -15,25 +15,28 @@ phases:
       - echo Build started on `date`
       - source ./.github/workflows/ci-helpers.sh
       - echo Build Sandbox Docker stage1
-      - pull_docker_cache $CACHE_REPO_NAME:$BUILDER_TAG $CACHE_REPO_NAME:$BUILDER_TAG0
+      - docker pull $CACHE_REPO_NAME:$BUILDER_TAG || docker pull $CACHE_REPO_NAME:$BUILDER_TAG0
       - |
         docker build \
           --target env_builder \
           --cache-from $CACHE_REPO_NAME:$BUILDER_TAG \
+          --cache-from $CACHE_REPO_NAME:$BUILDER_TAG0 \
           --tag        $CACHE_REPO_NAME:$BUILDER_TAG \
           ./docker/
       - echo push stage1 to cache repo
       - docker push $CACHE_REPO_NAME:$BUILDER_TAG
       - echo Build Sandbox Docker stage2
-      - pull_docker_cache $CACHE_REPO_NAME:$MAIN_TAG $CACHE_REPO_NAME:latest
+      - docker pull $CACHE_REPO_NAME:$MAIN_TAG || docker pull $CACHE_REPO_NAME:latest
       - |
         docker build \
+          --build-arg BUILDKIT_INLINE_CACHE=1
           --cache-from $CACHE_REPO_NAME:$BUILDER_TAG \
           --cache-from $CACHE_REPO_NAME:$MAIN_TAG \
-          --tag        $IMAGE_REPO_NAME:$MAIN_TAG \
+          --cache-from $CACHE_REPO_NAME:latest \
```

Comment:
Are you sure this will not pull both caches from dockerhub?
`pull_docker_cache` makes sure that if fallback is needed it gets re-tagged as original, in your case it won't.
If you are having issues `sourcing bash script`, how about taking out that function into a separate bash file and just running it normally.

Reply:
I'm reasonably confident that this doesn't pull from the Hub, it just uses them if they're around locally.

Reply:
Yep I tested it and it doesn't pull from dockerhub